runners/direct-java/src/test/java/org/apache/beam/runners/direct/DirectRunnerTest.java

```diff
@@ -577,4 +627,11 @@ public void validate() {
       return underlying.getOutputCoder();
     }
   }
+
+  /**
+   * Variable holder for each test instance.
+   */
+  public static class Context {
+    private volatile boolean teardown;
```

Comment:
Use `AtomicBoolean`;

Reply:
the usage is really set in one thread and read in another one so it is the same, but ok, can update it